db/migrate/20151009153646_thumbs_up_migration.rb

```diff
@@ -0,0 +1,25 @@
+class ThumbsUpMigration < ActiveRecord::Migration
+  def self.up
+    create_table :votes, :force => true do |t|
+
+      t.boolean    :vote,     :default => false,    :null => false
+      t.references :voteable, :polymorphic => true, :null => false
+      t.references :voter,    :polymorphic => true
+      t.timestamps
+
+    end
+
+    add_index :votes, [:voter_id, :voter_type]
+    add_index :votes, [:voteable_id, :voteable_type]
+
+
+    # Comment out the line below to allow multiple votes per voter on a single entity.
+    add_index :votes, [:voter_id, :voter_type, :voteable_id, :voteable_type], :unique => true, :name => 'fk_one_vote_per_user_per_entity'
+
+  end
+
+  def self.down
+    drop_table :votes
+  end
+
+end
```

Comment:
Look for this 'No newline at end of file' symbol that github shows you in all the files and add a newline where needed. https://robots.thoughtbot.com/no-newline-at-end-of-file